lib/mongoid/includes/inclusion.rb

```diff
@@ -17,6 +17,12 @@ def nested?
         !!from
       end
 
+      # Public: Checks if the collection already has an inclusion with the
+      # specified metadata.
+      def eql?(other)
+        metadata == other && other.respond_to?(:from) && from == other.from
```

Comment:
I think this condition should be:
```ruby
metadata == other && (!other.respond_to?(:from) || from == other.from)
```
Else it will allow duplicate additions to the inclusions `Set`, which we would want to prevent.

Can you add a test that fails when adding the same inclusion twice:
```ruby
    context 'prevents duplicates' do
      Given {
        inclusions.add(Band.relations['albums'])
      }
      Then { inclusions.size == 3 }
    end
```
Let's just add it in `inclusions_spec.rb`. Thanks!